deep_qa/models/sentence_selection/siamese_sentence_selector.py

```diff
@@ -56,18 +65,36 @@ def _build_model(self):
         # shape: (batch size, num_sentences, num_sentence_words, embedding size)
         sentences_embedding = self._embed_input(sentences_input)
 
-        # We encode the question embeddings with some encoder.
+        # We encode the question embedding with some more seq2seq layers
+        modeled_question = question_embedding
+        for i in range(self.num_seq2seq_layers):
+            hidden_layer = self._get_seq2seq_encoder(name="question_seq2seq_{}".format(i),
```

Comment:
I don't have a good intuition for whether it'd be better to share them or not.  Sharing them means fewer parameters, allowing them to be different lets you encode the query differently than you encode the corpus, which might be beneficial.  Allowing options is probably fine.  And I don't think this particular option would be that bad - just have a flag that called something like `share_hidden_layers`, and if it's true, use `seq2seq_{}` as the name, instead of `question_seq2seq_{}` and `passage_seq2seq_{}`.